### PR TITLE
[CDC-51] Expose CDC IO API

### PR DESCRIPTION
**Pull request by GiovanniMancini**
This PR proposes to expose APIs needed to request "Carta della Cultura" via APP IO.

### List of changes

- A new swagger template which represents exposed APIs
- A new API module needed to expose APIs on APIM 

### Motivation and context

This PR solves the problem of having an agreed upon interface agreement between cstar and IO APP to manage the request phase of "Carta della Cultura"

### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

---

### CODE REVIEW

File: src/api/cdc/openapi.io.yaml.tpl [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    post:
+      tags:
+      - /secured/utente
+      summary: 'Richiesta Carta della Cultura da prte del beneficiario'
+      operationId: registraBeneficiario
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/AnniRiferimento'
```
Comment (debiff) [inline src/api/cdc/openapi.io.yaml.tpl:43]: The `AnniRiferimento` should be required?

File: src/api/cdc/openapi.io.yaml.tpl [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+      type: object
+      properties:
+        listaStatoPerAnno:
+          type: array
+          items:
+            $ref: '#/components/schemas/StatoBeneficiarioPerAnno'
+    ListaEsitoRichiestaPerAnno:
+      type: object
+      properties:
+        listaEsitoRichiestaPerAnno:
```
Comment (debiff) [inline src/api/cdc/openapi.io.yaml.tpl:161]: The `listaEsitoRichiestaPerAnno` should be required?

**Review by debiff — CHANGES_REQUESTED** [review]

File: src/api/cdc/openapi.io.yaml.tpl (hunk 2 above)
Comment (GiovanniMancini) [reply src/api/cdc/openapi.io.yaml.tpl:161]: I agree. I'll fix it

**Review by debiff — DISMISSED** [review]
The swagger LGTM

File: src/api/cdc/openapi.io.yaml.tpl [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        - INATTIVO # carta richiesta e risposta negativa da INPS
+    EsitoRichiesta:
+      type: string
+      enum: 
+        - INIZIATIVA_TERMINATA
+        - ANNO_NON_AMMISSIBILE
+        - CIT_REGISTRATO
+    Anno: 
+      type: string 
+      pattern: '^\d{4}$'
```
Comment (debiff) [inline src/api/cdc/openapi.io.yaml.tpl:93]: Can you add a double backslash?
```suggestion
      pattern: '^\\d{4}$'
```
Unfortunately on the app we are using an old version of `openapi-codegen-ts` and we suffer from this problem.

**Review by debiff — CHANGES_REQUESTED** [review]

File: src/api/cdc/openapi.io.yaml.tpl [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+      type: string
+      enum: 
+        - ATTVABILE
+        - INATTIVABILE 
+        - ATTIVO # carta richiesta e risposta ok da INPS
+        - VALUTAZIONE # carta richiesta ma non si conosce la risposta da INPS
+        - INATTIVO # carta richiesta e risposta negativa da INPS
+    EsitoRichiesta:
+      type: string
+      enum:
```
Comment (debiff) [inline src/api/cdc/openapi.io.yaml.tpl:87]: Among the values of `EsitoRichiesta` there should be also the success value. Do you agree?

**Review by debiff — CHANGES_REQUESTED** [review]

File: src/api/cdc/openapi.io.yaml.tpl [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+          $ref: '#/components/schemas/StatoBeneficiario'
+        annoRiferimento:
+          $ref: '#/components/schemas/Anno'
+    EsitoRichiestaPerAnno:
+      type: object
+      required:
+        - statoBeneficiario
+        - annoRiferimento
+      properties:
+        statoBeneficiario:
```
Comment (GiovanniMancini) [inline src/api/cdc/openapi.io.yaml.tpl:148]: Rinominare property in `esitoRichiesta`

File: src/api/cdc/openapi.io.yaml.tpl (hunk 4 above)
Comment (GiovanniMancini) [reply src/api/cdc/openapi.io.yaml.tpl:87]: Yes, I agree. Let's call it `OK`

File: src/api/cdc/openapi.io.yml.tpl [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+components:
+  securitySchemes:
+    BearerAuth:
+      type: http
+      scheme: bearer
+  schemas:
+    StatoBeneficiario:
+      type: string
+      enum: 
+        - ATTVABILE
```
Comment (debiff) [inline src/api/cdc/openapi.io.yml.tpl:80]:
```suggestion
        - ATTIVABILE
```

**Review by debiff — CHANGES_REQUESTED** [review]

**Pull request merged**